### DISCUSSION TITLE
Allow creating a table with a bigint primary key

### DISCUSSION THREAD

**Discussion by sxsws** [directus/directus#13290] [Feature Requests]
Sometimes a lot of data is envisioned, or has to go in as lookup data and it'd be nice to be able to create a collection with 'index with bigint' alongside the numerical pk, guid and manual entry helper box as it isn't possible to change the type after field creation.

**Comment by benhaynes** [comment]
You're looking to store more than 4.3 billion items in a table? 🤔 You can probably still do this, but in the database directly. Not sure if this would pass our 80/20 rule.

**Comment by samdze** [comment]
I think this is important.
Overcoming the integer limit is not incredibly hard when you consider deleted items, M2M tables where items are constantly added and deleted, or generic SaaS environments with multi-tenancy and thousands of users.

**Reply by benhaynes** [reply]
Do you feel this is (80/20) important to configure through the App? Any reason changing/creating the PK in the database directly isn't a sufficient escape hatch?

**Reply by samdze** [reply]
Well, I _feel_ this might be such an easy and harmless addition to the Directus web app that I don't see why it shouldn't be provided as a feature.
I certainly would use it more than other built-in features.
The need to change the PK type directly in the database feels unnecessary and a missed opportunity to me.

**Reply by sxsws** [reply]
Exactly as above, I would have thought it would be an easy addition and when pulling / scraping data (e.g. product data from multiple suppliers) with countless variations the numbers can quickly tot up.

**Reply by benhaynes** [reply]
Fair enough! We'll track the feature request upvotes over time to gauge community support, in the meantime, we welcome any coordinated PRs or sponsorings.

**Reply by samdze** [reply]
I'm going to give it a shot.
Feels like a good opportunity to get more familiar with the codebase!

**Reply by sxsws** [reply]
i've been using directus for about 30 hours so I'm still finding my feet!

**Comment by samdze** [comment]
Created a PR 🎉 https://github.com/directus/directus/pull/13301

**Reply by benhaynes** [reply]
Amazing! Thank you!!

**Reply by sxsws** [reply]
Now there's a PR, do I close this discussion? Can I do that? Not sure how github discussions work!

**Reply by benhaynes** [reply]
Once the pull request is merged, we'll simply change the category of this discussion to implemented.
